inc/saitam.h

```diff
@@ -1416,6 +1419,37 @@ typedef enum _sai_tam_report_attr_t
      */
     SAI_TAM_REPORT_ATTR_REPORT_INTERVAL_UNIT,
 
+    /**
+     * @brief Sampling rate (every 1/sample_rate)
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @condition SAI_TAM_REPORT_ATTR_REPORT_MODE == SAI_TAM_REPORT_MODE_SAMPLING
+     */
+    SAI_TAM_REPORT_ATTR_SAMPLE_RATE,
+
+    /**
+     * @brief Maximum report rate per second
+     *
+     * Value 0 to no limit.
+     *
+     * @type sai_uint64_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     * @validonly SAI_TAM_REPORT_ATTR_REPORT_MODE == SAI_TAM_REPORT_MODE_ALL
+     */
+    SAI_TAM_REPORT_ATTR_MAX_REPORT_RATE,
+
+    /**
+     * @brief Maximum reports per burst
+     *
+     * @type sai_uint64_t
+     * @flags CREATE_AND_SET
+     * @default 0
```

Comment:
What would be the behavior for a default value of 0? Is the implementation expected to sent out zero reports?

Reply:
If value is 0, it means no rate-limit and all reports will be sent out.

Reply:
I understand the behavior for the MAX_REPORT_RATE attribute and there is a comment there as well. But I'm still unclear on a default value of 0 for MAX_REPORT_BURST.

Let's say MAX_REPORT_RATE is set to 1000 reports/second and MAX_REPORT_BUTST is not set so it defaults to 0. What will be the expected behavior? Is the device expected to ignore the max_report_rate and sent out all reports?

Reply:
In this case, the device should not ignore the MAX_REPORT_RATE. With MAX_REPORT_BURST set to 0, no bursts are allowed, so the transmission rate of reports will be limited to 1000 reports per second. If a burst of traffic is received, it depends on the vendor's implementation whether the excess reports are dropped or delayed.

Reply:
When MAX_REPORT_RATE = 0, rate-limiting is not applicable. With this is, MAX_REPORT_BURST defaulted to value 1 seems incorrect.
To align with existing SAI model (similar to shaper), we can have MAX_REPORT_BURST default as 0 and it's up to the user to configure MAX_REPORT_RATE and MAX_REPORT_BURST with vendor supported values for the rate-limiter to work.

Reply:
Hi Raj, I was asking about the case when MAX_REPORT_RATE is non-zero and MAX_REPORT_BURST is not defined. 

But I'm fine with the user properly defining these two values to make the rate-limiter work properly.